block/manager_test.go

```diff
@@ -227,6 +227,48 @@ func TestSubmitBlocksToDA(t *testing.T) {
 			isErrExpected:               true,
 			expectedPendingBlocksLength: 1,
 		},
+		{
+			name: "A and B are fair blocks, but C has a marshalling error. So C never gets submitted",
+			blocks: func() []*types.Block {
+				numBlocks, numTxs := 3, 5
+				blocks := make([]*types.Block, numBlocks)
+				blocks[0] = types.GetRandomBlock(uint64(1), numTxs)
+				blocks[1] = types.GetRandomBlock(uint64(2), numTxs)
+				blocks[2] = types.GetRandomBlock(uint64(3), numTxs)
+				invalidateBlockHeader(blocks[2])
+				return blocks
+			}(),
+			isErrExpected:               true,
+			expectedPendingBlocksLength: 1,
```

Comment:
```suggestion
			name: "A and B are fair blocks, but C has a marshalling error which results in no blocks being submitted.",
			blocks: func() []*types.Block {
				numBlocks, numTxs := 3, 5
				blocks := make([]*types.Block, numBlocks)
				blocks[0] = types.GetRandomBlock(uint64(1), numTxs)
				blocks[1] = types.GetRandomBlock(uint64(2), numTxs)
				blocks[2] = types.GetRandomBlock(uint64(3), numTxs)
				invalidateBlockHeader(blocks[2])
				return blocks
			}(),
			isErrExpected:               true,
			expectedPendingBlocksLength: 3,
```

Reply:
still needs addressing plus it looks like their is a conflict with main now that needs to be addressed @arhamj